### ISSUE TITLE
Stop Seeding during runtime

### ISSUE THREAD

**Issue by gitpubber** [anacrolix/torrent#479]
How can I do it? There is ClientConfig.Seed but how can I change it during runtime ?

**Comment by anacrolix** [comment]
This will need to be exposed. It probably wouldn't hurt to expose stuff like download/upload ratio too to enable users to make use of that in configuring seeding.

**Comment by gitpubber** [comment]
Can you please expose seed and download/upload ratio ?

**Comment by anacrolix** [comment]
I will look into it if I have time, cheers.

**Comment by gitpubber** [comment]
Hello, what's the progress on this? also, to change iplist.Ranger , should we lock client's mutex before changing? if yes, can you please create wrapper function to perform so?

**Comment by anacrolix** [comment]
Sorry, I'm not working on this currently.

**Comment by anacrolix** [comment]
This remains a valid enhancement. Possibly related to https://github.com/anacrolix/torrent/issues/155.

**Comment by gitpubber** [comment]
@anacrolix ,
Possible workarounds include ranging through torrents in client and if they are completed, disable data upload

**Comment by anacrolix** [comment]
Yeah, there are definitely workarounds. Good thinking.

**Comment by gitpubber** [comment]
https://github.com/anacrolix/torrent/blob/b21aebeaae6a308d97e9f952c78639bc594e73fd/torrent.go#L1363

So, Config is directly been referenced. so,it would be easy to implement this feature,no?

**Comment by gitpubber** [comment]
@anacrolix any update on this?

**Comment by gitpubber** [comment]
@anacrolix can you provide method to change Seed value of config(while mutex locking it) ?

It must solve this issue.right?

**Comment by anacrolix** [comment]
Yes that would work. There might be a few condition values to tickle or something too.

**Comment by gitpubber** [comment]
https://github.com/anacrolix/torrent/pull/527 @anacrolix
